lib/sentry/context.ex

```diff
@@ -40,26 +61,129 @@ defmodule Sentry.Context do
     }
   end
 
+  @doc """
+  Merges new fields into the `:extra` context, specific to the current process.
+
+  This is used to set fields which should display when looking at a specific
+  instance of an error.
+
+  ## Example
+
+      iex> Sentry.Context.set_extra_context(%{id: 123})
+      nil
+      iex> Sentry.Context.set_extra_context(%{detail: "bad_error"})
+      %{extra: %{id: 123}}
+      iex> Sentry.Context.set_extra_context(%{message: "Oh no"})
+      %{extra: %{detail: "bad_error", id: 123}}
+      iex> Sentry.Context.get_all()
+      %{
+        user: %{},
+        tags: %{},
+        extra: %{detail: "bad_error", id: 123, message: "Oh no"},
+        request: %{},
+        breadcrumbs: []
+      }
+  """
+  @spec set_extra_context(map()) :: nil | map()
   def set_extra_context(map) when is_map(map) do
     get_context()
     |> set_context(@extra_key, map)
   end
 
+  @doc """
+  Merges new fields into the `:user` context, specific to the current process.
+
+  This is used to set certain fields which identify the actor who experienced a
+  specific instance of an error.
```

Comment:
I haven't actually used the `:user` context myself, so feel free to suggest alterations if what I've said isn't accurate or comprehensive enough here!